### ISSUE TITLE
Clicking on a link in a Flutter Web page using an accessibility service causes the page to reload

### ISSUE THREAD

**Issue by math1man** [flutter/flutter#146291]
### Steps to reproduce

1. Copy the repro project into DartPad and run it
2. Enable accessibility in Flutter
3. Use an accessibility service or a mouse to click on the button

### Expected results

Nothing should happen, similar to clicking on the button with a mouse without enabling accessibility.

### Actual results

The UI performs a screen transition as if it were some sort of navigation. I believe this symptom is also what is causing the reload in our app.

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      home: Scaffold(
        body: Center(
          child: InkWell(
            onTap: () {},
            child: Semantics(
              link: true,
              child: const Text('Hello, World!'),
            ),
          ),
        ),
      ),
    );
  }
}
```

</details>

### Flutter Doctor output

Internal bug b/328226356

**Comment by darshankawar** [comment]
Thanks for the bug report @math1man 
I tried the code sample using latest stable (3.19.5) running on web with accessibility enabled but was unable to see the page getting reloaded. 


https://github.com/flutter/flutter/assets/67046386/a77e1460-93de-4110-bfe7-16288e7b831e



> Internal bug b/328226356

I can't access it, so not sure which flutter version are you on. If above doesn't represent correct reproduction, can you please elaborate or provide short video of the behavior ?

**Comment by math1man** [comment]
I reproduced the issue specifically in DartPad. Here's a video:

https://github.com/flutter/flutter/assets/4040946/6aca9b69-6a2a-4fb5-ac48-35408cf7e8af

**Comment by math1man** [comment]
I also forgot to mention, but removing this one line fixes the issue:

https://github.com/flutter/engine/blob/beee02b5ba2d2ee45b99ff7ae32f75175aa67821/lib/web_ui/lib/src/engine/semantics/link.dart#L23

**Comment by peixinli** [comment]
Update the priority to match the internal issue.

**Comment by mdebbar** [comment]
I'm actively working on this and https://github.com/flutter/flutter/issues/143164.
